crates/ty_python_semantic/src/types/ide_support.rs

```diff
@@ -633,11 +618,11 @@ pub fn definitions_for_name<'db>(
             return union
                 .elements(db)
                 .iter()
+                .rev()
                 .filter_map(|ty| ty.as_nominal_instance())
                 .map(|instance| {
                     let definition = instance.class_literal(db).definition(db);
-                    let parsed = parsed_module(db, definition.file(db));
-                    ResolvedDefinition::FileWithRange(definition.focus_range(db, &parsed.load(db)))
```

Comment:
I realized right after landing the *go to definition for float and complex* PR that using `FileWithRange` here is a stupid idea. I switched to using `FileWithRange` to suppress the docstring (we always pick the first docstring which ended up to be `int` which is rather confusing). However, returning `FileWithRange` here breaks semantic highlighting because a `FileWithRange` has no definition.

That's why I opted in this PR to keep using `Definition` but return the definition for `float` first (or `complex`) so that we show the docstring of `float`. 

This is not 100% correct because `float` is a union in this position, but it also doesn't feel entirely wrong (and sort of what I'd expect as a user).